app/models/movie.rb

```diff
@@ -1,3 +1,17 @@
 class Movie < ApplicationRecord
   belongs_to :studio
+  has_many :actor_movies
+  has_many :actors, through: :actor_movies
+
+  def actors_by_age
+    actors.order(:age).pluck(:name)
+  end
+
+  def average_actor_age
+    actors.average(:age)
+  end
+
+  def add_actor(actor_id)
```

Comment:
This is fine but not necessary - we can do things like `<<` and `find`/`find_by` in the controller without violating MVC concerns. The other reason I would avoid this is that now you have to _test it_; but when you look at it, what are you testing? Really just the `<<` operator, and that comes standard with Ruby, so it's kinda overkill. I like the effort though!